File: backend/app/schemas/risk.py
from pydantic import BaseModel, ConfigDict, Field


class RiskOverview(BaseModel):
    highRiskCount: int
    mediumRiskCount: int
    lowRiskCount: int
    avgRiskScore: float

    model_config = ConfigDict(from_attributes=True)


class RiskHistogramBin(BaseModel):
    label: str
    start: float
    end: float
    count: int

    model_config = ConfigDict(from_attributes=True)


class RiskSegmentationItem(BaseModel):
    riskLevel: str
    supplierCount: int

    model_config = ConfigDict(from_attributes=True)


class RiskSupplierItem(BaseModel):
    supplierId: int
    supplierName: str
    country: str | None = None
    category: str | None = None
    avgDelay: float
    avgDefect: float
    avgCostVariance: float
    riskScore: float
    riskLevel: str

    model_config = ConfigDict(from_attributes=True)


class DueDiligenceRequest(BaseModel):
    supplierId: int = Field(gt=0)


class DueDiligenceResponse(BaseModel):
    supplier: str
    opRisk: str
    esgRisk: str
    overall: str
    issues: list[str]
    aiSummary: str

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/risk.ts
import { apiRequest } from "./client";

export interface RiskOverview {
  highRiskCount: number;
  mediumRiskCount: number;
  lowRiskCount: number;
  avgRiskScore: number;
}

export interface RiskHistogramBin {
  label: string;
  start: number;
  end: number;
  count: number;
}

export interface RiskSegmentationItem {
  riskLevel: "High" | "Medium" | "Low";
  supplierCount: number;
}

export interface RiskSupplierItem {
  supplierId: number;
  supplierName: string;
  country: string | null;
  category: string | null;
  avgDelay: number;
  avgDefect: number;
  avgCostVariance: number;
  riskScore: number;
  riskLevel: "High" | "Medium" | "Low";
}

export interface DueDiligenceResponse {
  supplier: string;
  opRisk: string;
  esgRisk: string;
  overall: string;
  issues: string[];
  aiSummary: string;
}

export async function getRiskOverview(): Promise<RiskOverview> {
  return apiRequest<RiskOverview>("/risk/overview");
}

export async function getRiskDistribution(): Promise<RiskHistogramBin[]> {
  return apiRequest<RiskHistogramBin[]>("/risk/distribution");
}

export async function getRiskSegmentation(): Promise<RiskSegmentationItem[]> {
  return apiRequest<RiskSegmentationItem[]>("/risk/segmentation");
}

export async function getTopRiskSuppliers(): Promise<RiskSupplierItem[]> {
  return apiRequest<RiskSupplierItem[]>("/risk/top-suppliers");
}

export async function runDueDiligence(supplierId: number): Promise<DueDiligenceResponse> {
  return apiRequest<DueDiligenceResponse>("/risk/due-diligence", {
    method: "POST",
    json: { supplierId },
  });
}
